xgcm/grid.py

```diff
@@ -284,12 +293,22 @@ def _get_neighbor_data_pairs(self, da, position_to, boundary=None,
             left = da.data
         elif (self._periodic and ((transition == ('center', 'left')) or
                                   (transition == ('right', 'center')))):
-            left = da.roll(**{dim: 1}).data
+
+            if self._wrap is not None:
+                left = da.roll(**{dim: 1})
+                left = add_to_slice(left, dim, 0, -self._wrap).data
+            else:
+                left = da.roll(**{dim: 1}).data
```

Comment:
can we put this wrap logic in standalone function?

Reply:
I think yes, but I would probably have to repeat all the positional logic. I did sinplify the wrap logic, soo maybe that is acceptable?